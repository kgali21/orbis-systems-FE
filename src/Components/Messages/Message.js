import React from 'react';
import PropTypes from 'prop-types';
import styles from './Message.module.css';

const Message = ({ tweets, timeStamp }) => (
    <div className={styles.Tweets}>
        <p>{tweets}</p>
        <p className={styles.TimeStamp}>{timeStamp}</p>
    </div>
)

Message.propTypes = {
    tweets: PropTypes.string.isRequired,
    timeStamp: PropTypes.string.isRequired
}

export default Message;