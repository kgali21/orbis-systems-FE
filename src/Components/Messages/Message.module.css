.Tweets {
    font-size: .75em;
}