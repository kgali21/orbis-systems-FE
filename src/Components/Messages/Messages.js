import React from 'react';
import Message from './Message';

const Messages = ({ messages }) => {
    const messageElements = messages.map((data, i) => 
        <li key={i}>
            <Message tweets={data.body} timeStamp={data.created_at}/>
        </li>
    );

    return (
            <ul>
                {messageElements}
            </ul>
    );


};

export default Messages;