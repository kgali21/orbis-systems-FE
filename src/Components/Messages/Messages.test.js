import React from 'react';
import { shallow } from 'enzyme';
import Messages from './Messages';

describe('Messages Component', () => {
    it('renders messages', () => {
        const messages = [
            {
                body: 'test',
                created_at: '12:45'
            }
        ]

        const wrapper = shallow(<Messages messages={messages} />);
        expect(wrapper).toMatchSnapshot();
    });
});