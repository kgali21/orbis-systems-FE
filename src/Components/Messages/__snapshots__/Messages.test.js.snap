// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages Component renders messages 1`] = `ShallowWrapper {}`;
